### PR TITLE
more tests(sub, pub qos0, unsub) for broker_tcp and nmq_mqtt

### PR DESCRIPTION
**Pull request by Hermann0222**

### CODE REVIEW

**Review by JaylinYu — APPROVED** [review]
thankssssssssssssssssssssssssss

**Comment by JaylinYu** [comment]
whoops aio appending error

**Review by lee-emqx — APPROVED** [review]

**Comment by Hermann0222** [comment]
> whoops aio appending error

Can you describe the problem in more detail?
I don't quiet get it.

File: tests/trantest.h [hunk 1]
```diff
@@ -745,16 +1126,62 @@ trantest_mqtt_broker_send_recv(trantest *tt)
...
+		nng_ctx_recv(work->ctx, work->aio);
+		So((rmsg = nng_aio_get_msg(work->aio)) != NULL);
+		So(nng_msg_get_type(rmsg) == CMD_UNSUBSCRIBE);
+		So((work->unsub_pkt = nng_alloc(sizeof(packet_unsubscribe))) != NULL);
+		work->msg = rmsg;
+		So(decode_unsub_msg(work) == 0);
+		So(encode_unsuback_msg(rmsg, work) == 0);
+		nng_aio_set_msg(work->aio, rmsg);
+		nng_ctx_send(work->ctx, work->aio);
+		nng_aio_finish(work->aio, 0);
```
Comment (wanghaEMQ) [inline tests/trantest.h:1175]: Why? It should be finished by nng?

Comment (wanghaEMQ) [reply tests/trantest.h:1175]: > whoops aio appending error

I think the error of aio appending was caused in this line

**Comment by JaylinYu** [comment]
bravo you have done it!

**Review by JaylinYu — APPROVED** [review]

File: tests/trantest.h (hunk 1 above)
Comment (JaylinYu) [reply tests/trantest.h:1175]: perhaps you are right
but we can fix in another PR

**Pull request merged**